js/modules/types/message.js

```diff
@@ -157,8 +157,17 @@ exports._mapQuotedAttachments = upgradeAttachment => async (message, context) =>
     return message;
   }
 
-  const upgradeWithContext = attachment =>
-    upgradeAttachment(attachment, context);
+  const upgradeWithContext = async (attachment) => {
+    if (!attachment || !attachment.thumbnail) {
```

Comment:
Is a `false`-y attachment a valid input? If so, when does that happen? If not, could we `throw` a `TypeError`?

Reply:
This should never happen, for sure. I still don't like the idea of throwing an exception if we do get a falsey attachment. Why? Because 1) the message just wouldn't show up and 2) there's no guarantee we'll get a bug report from a user.

In terms of cleanliness, perhaps the right thing to do is a `compact()` on the array so we really do eliminate any problematic elements of the array.